modules/fedimint-walletv2-client/src/lib.rs

```diff
@@ -0,0 +1,570 @@
+#![deny(clippy::pedantic)]
+#![allow(clippy::missing_errors_doc)]
+#![allow(clippy::missing_panics_doc)]
+#![allow(clippy::must_use_candidate)]
+#![allow(clippy::module_name_repetitions)]
+
+use std::collections::{BTreeMap, BTreeSet};
+use std::fmt::Debug;
+
+use api::WalletFederationApi;
+use bitcoin::address::NetworkUnchecked;
+use bitcoin::{Address, Network};
+use db::AddressCounterKey;
+use fedimint_api_client::api::{DynModuleApi, FederationResult};
+use fedimint_client::db::ClientMigrationFn;
+use fedimint_client::derivable_secret::{ChildId, DerivableSecret};
+use fedimint_client::module::init::{ClientModuleInit, ClientModuleInitArgs};
+use fedimint_client::module::recovery::NoModuleBackup;
+use fedimint_client::module::{ClientContext, ClientModule, IClientModule};
+use fedimint_client::sm::{Context, DynState, State, StateTransition};
+use fedimint_client::transaction::{
+    ClientInput, ClientInputBundle, ClientOutput, ClientOutputBundle, TransactionBuilder,
+};
+use fedimint_client::DynGlobalClientContext;
+use fedimint_core::core::{Decoder, IntoDynInstance, ModuleInstanceId, ModuleKind, OperationId};
+use fedimint_core::db::{
+    Database, DatabaseTransaction, DatabaseVersion, IDatabaseTransactionOpsCoreTyped,
+};
+use fedimint_core::encoding::{Decodable, Encodable};
+use fedimint_core::module::{
+    ApiVersion, CommonModuleInit, ModuleCommon, ModuleInit, MultiApiVersion,
+};
+use fedimint_core::util::SafeUrl;
+use fedimint_core::{apply, async_trait_maybe_send, Amount, OutPoint, TransactionId};
+use fedimint_walletv2_common::config::WalletClientConfig;
+use fedimint_walletv2_common::esplora_api::{DynEsploraConnection, RealEsploraConnection};
+use fedimint_walletv2_common::{
+    descriptor, DestinationScript, ReceiveFee, SendFee, WalletCommonInit, WalletInput,
+    WalletInputV0, WalletModuleTypes, WalletOutput, WalletOutputV0,
+};
+use secp256k1::Keypair;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+mod api;
+mod cli;
+mod db;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum WalletOperationMeta {
+    Send(SendMeta),
+    Receive(ReceiveMeta),
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SendMeta {
+    pub txid: TransactionId,
+    pub change_outpoints: Vec<OutPoint>,
+    pub address: Address<NetworkUnchecked>,
+    pub amount: bitcoin::Amount,
+    pub fee: bitcoin::Amount,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ReceiveMeta {
+    pub txid: TransactionId,
+    pub change_outpoints: Vec<OutPoint>,
+    pub amount: bitcoin::Amount,
+    pub fee: bitcoin::Amount,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub enum FinalOperationState {
+    /// The operation has been successful.
+    Success,
+    /// The operation has been aborted due to a change of the consensus feerate.
+    Aborted,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct UnspentDeposit {
```

Comment:
Unspent == Unclaimed? And here it's a Deposit, but above "Receive"? :D

`DepositMeta` seems to make more sense than `ReceiveMeta` in that case. The operations can be "send" and "receive" but when talking about nouns, "deposit" and "withdrawal" sound better, IMO.

Reply:
Writting docstring for these would clarify any doubts and help with the naming.